src/GrafanaDashboardDeploy/src/GrafanaClient.cs

```diff
@@ -0,0 +1,327 @@
+            HttpResponseMessage response = await _client.GetAsync(uri).ConfigureAwait(false);
+
+            Health health = null;
+
+            if (response.IsSuccessStatusCode)
+            {
+
+                using (Stream stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false))
+                using (var streamReader = new StreamReader(stream))
+                using (var jsonReader = new JsonTextReader(streamReader))
+                {
+                    health = JsonSerializer.CreateDefault().Deserialize<Health>(jsonReader);
+                }
+            }
+            else
+            {
+                _logger.LogWarning("Response does not indicate success");
+            }
+
+            return health;
+        }
+
+        public async Task<JObject> GetDashboardAsync(string uid)
+        {
+            var uri = new Uri(new Uri(BaseUrl), $"/api/dashboards/uid/{uid}");
+            JObject dashboard = new JObject();
+
+            HttpResponseMessage response = await _client.GetAsync(uri).ConfigureAwait(false);
+
+            if (response.IsSuccessStatusCode)
+            {
+
+                using (Stream stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false))
+                using (var streamReader = new StreamReader(stream))
+                using (var jsonReader = new JsonTextReader(streamReader))
+                {
+                    dashboard = await JObject.LoadAsync(jsonReader).ConfigureAwait(false);
+                }
+            }
+            else
+            {
+                _logger.LogWarning("Response does not indicate success");
+                _logger.LogWarning("Status {}: {}", response.StatusCode, response.ReasonPhrase);
+                _logger.LogDebug(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+            }
+
+            return dashboard;
+        }
+
+        /// <summary>
+        /// Get a Grafana Folder by its uid
+        /// </summary>
+        /// <param name="uid">The folder uid</param>
+        /// <returns>The Folder JSON object</returns>
+        public async Task<JObject> GetFolder(string uid)
+        {
+            var uri = new Uri(new Uri(BaseUrl), $"/api/folders/{uid}");
+            var folder = new JObject();
+
+            HttpResponseMessage response = await _client.GetAsync(uri).ConfigureAwait(false);
+
+            if (response.IsSuccessStatusCode)
+            {
+
+                using (Stream stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false))
+                using (var streamReader = new StreamReader(stream))
+                using (var jsonReader = new JsonTextReader(streamReader))
+                {
+                    folder = await JObject.LoadAsync(jsonReader).ConfigureAwait(false);
+                }
+            }
+            else
+            {
+                _logger.LogWarning("Response does not indicate success");
```

Comment:
You can proabably call "response.EnsureSuccessful()" and handle the exception it's going to throw higher (so it doesn't have to be repeated everywhere), and wrap it in some sort of retry policy.